### PR TITLE
Add Report Bug and Inform Direction

### PR DESCRIPTION
**Pull request by aneelac22**
Screen Shot 2022-10-12 at 8 06 33 AM
Screen Shot 2022-10-12 at 8 06 44 AM
Screen Shot 2022-10-12 at 8 07 09 AM
Screen Shot 2022-10-12 at 8 07 48 AM
Screen Shot 2022-10-12 at 8 07 57 AM
[Jira Ref RHCLOUD-20863]

Waiting UX review on one screen

### CODE REVIEW

File: src/js/App/Feedback/InformDirection.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        </Button>
+        <Button ouiaId="cancel-feedback" className="chr-c-feedback-footer-button" key="cancel" variant="link" onClick={onCloseModal}>
+          {intl.formatMessage(messages.cancel)}
+        </Button>
+      </div>
+    </div>
+  );
+};
+
+InformDirection.propTypes = {
```
Comment (Hyperkid123) [inline src/js/App/Feedback/InformDirection.tsx:103]: No need for prop types in TS files.

File: src/js/App/Feedback/InformDirection.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+          Authorization: `Bearer ${token}`,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          description: `[Research Opportunities] Username: ${user.identity.user.username}, Account ID: ${user.identity.account_number}, 
+          Email: ${checked ? user.identity.user.email : ''}, URL: ${window.location.href}`,
+          summary: `${addFeedbackTag()} App Feedback`,
+          labels: [app, bundle],
+        }),
+      }).then((response) => response.json());
```
Comment (Hyperkid123) [inline src/js/App/Feedback/InformDirection.tsx:43]: Any reason for parsing the response into JSON if we are not using it?

File: src/js/App/Feedback/ReportBug.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        </Button>
+        <Button ouiaId="cancel-feedback" className="chr-c-feedback-footer-button" key="cancel" variant="link" onClick={onCloseModal}>
+          {intl.formatMessage(messages.cancel)}
+        </Button>
+      </div>
+    </div>
+  );
+};
+
+ReportBug.propTypes = {
```
Comment (Hyperkid123) [inline src/js/App/Feedback/ReportBug.tsx:140]: ^^

File: src/js/App/Feedback/ReportBug.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  const app = getUrl('app');
+  const bundle = getUrl('bundle');
+  const isAvailable = env === 'prod' || env === 'stage';
+  const addFeedbackTag = () => (isProd() ? `[${bundle}]` : '[PRE-PROD]');
+
+  async function handleModalSubmission() {
+    const token = await window.insights.chrome.auth.getToken();
+    if (isAvailable) {
+      try {
+        const response = await fetch(`${window.origin}/api/platform-feedback/v1/issues`, {
```
Comment (Hyperkid123) [inline src/js/App/Feedback/ReportBug.tsx:46]: The fetch function seems to be always almost the same, only the body changes. Can we create one general function and configure it with arguments?

File: src/js/App/Feedback/ReportBug.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+      <TextContent>
+        <Text component={TextVariants.h1}>{intl.formatMessage(messages.reportABug)}</Text>
+        <Text>
+          {intl.formatMessage(messages.describeReportBug)}{' '}
+          <Text component="a" href="https://access.redhat.com/support/cases/#/case/new/open-case?caseCreate=true" target="_blank">
+            {intl.formatMessage(messages.openSupportCase)} <ExternalLinkAltIcon />
+          </Text>
+        </Text>
+      </TextContent>
+      <Form>
```
Comment (Hyperkid123) [inline src/js/App/Feedback/ReportBug.tsx:83]: MIght be overkill but we should probably use DDF for all the forms.

File: src/js/App/Feedback/ReportBug.tsx (hunk 4 above)
Comment (aneelac22) [reply src/js/App/Feedback/ReportBug.tsx]: I actually further refactored it

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**